src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -5,73 +5,77 @@
 namespace Microsoft.Rest
 {
     using System;
+    using System.Collections.Concurrent;
     using System.Collections.Generic;
     using System.Linq;
     using System.Reflection;
 
     /// <summary>
     /// Abstract base class for types that act like enums, but can be extended with arbitrary string values.
     /// </summary>
-    public abstract class ExtensibleEnum<T> : IEquatable<T> where T : ExtensibleEnum<T>
+    public abstract class ExtensibleEnum<T, V> : IEquatable<T> where T : ExtensibleEnum<T, V>
     {
-        private static readonly Lazy<Dictionary<string, T>> _nameMap =
-            new Lazy<Dictionary<string, T>>(CreateNameMap, isThreadSafe: true);
+        private static readonly ConcurrentDictionary<V, T> _valueMap =
```

Comment:
Given that this dictionary is never mutated after it's constructed (and IMO never, ever, should be), what does using `ConcurrentDictionary` buy you?

Also, I still haven't seen a good argument against lazy initialization here.

Reply:
Agreed. @olydis thoughts?